src/libxdata/XWorld.cpp

```diff
@@ -81,7 +81,7 @@ void XWorld::forEachAirport(std::function<void(std::shared_ptr<world::Airport>)>
     }
 }
 
-std::shared_ptr<world::Region> XWorld::findOrCreateRegion(const std::string& id) {
+std::shared_ptr<world::Region> XWorld::getRegion(const std::string& id) {
```

Comment:
Hm.. I think I prefer the old name as `getRegion` now has the odd side effect of creating the region if it doesn't exist. From a function called `get...`, I would expect failure if something doesn't exist. Now someone could use this function on user-facing data and unintentionally create new regions instead of showing an error.
As a compromise, maybe it makes sense to make `getRegion` return failure if the region doesn't exist and then implement `findOrCreateRegion` for the loader using the new function, only creating the region on failure. Then both functions would be visible and both could be used depending on the use case.

Reply:
Yes, good point. I changed the name to reflect the usage in the SQL-backed World implementation, but of course I haven't submitted that yet. I agree it is not a good idea to lose the original meaning for the XWorld variant. I'll revert to the previous name, and add the new interface as suggested.